packages/babel-parser/test/fixtures/typescript/class/optional-generator/input.ts

```diff
@@ -0,0 +1,9 @@
+class A {
+  *[a?.b]?() { };
```

Comment:
Should we report a recoverable error in cases like this, similarly to how report it for `readonly` and `declare`?

Reply:
Actually, this is valid TypeScript syntax!.

Do you mean `readonly *[a?.b]?() { };` ?

Reply:
Oh I meant with just `?`, I didn't realize it's actually valid syntax.